client/consensus/manual-seal/src/consensus/babe.rs

```diff
@@ -150,6 +186,43 @@ impl<B, C> ConsensusDataProvider<B> for BabeConsensusDataProvider<B, C>
 			.map_err(|e| Error::StringError(format!("failed to fetch epoch data: {}", e)))?
 			.ok_or_else(|| sp_consensus::Error::InvalidAuthoritiesSet)?;
 
+		let epoch = epoch_changes
+			.viable_epoch(
+				&epoch_descriptor,
+				|slot| Epoch::genesis(&self.config, slot),
+			)
+			.ok_or_else(|| {
+				log::info!(target: "babe", "create_digest: no viable_epoch :(");
+				sp_consensus::Error::InvalidAuthoritiesSet
+			})?;
+
+		// a quick check to see if we're in the authorities
+		let has_authority = epoch.as_ref()
+			.authorities
+			.iter()
+			.find(|(id, _)| {
+				*id == AuthorityId::from(Alice.public())
```

Comment:
Same here. For tests it would probably be useful to be able to configure authorities.